### DISCUSSION TITLE
Adapting the script for more than 1 inverter

### DISCUSSION THREAD

**Discussion by Skjal** [ZuinigeRijder/SolisCloud2PVOutput#23] [General]
Hi.

just got it up and running but I see in the description that you need to adjust the code if you have more than 1 inverter. I currently have 2 inverters and would like to get this working with both but not very good at scripting/coding so just wanted to ask if you can help me getting this sorted out. Thanks for sharing your code by the way.

**Comment by ZuinigeRijder** [comment]
@Skjal 
I have also 2 inverters, but not both from Solis, one Solis inverter and one SMA inverter. I have for each inverter a separate PVOutput account. Because I also wanted to have the combined PVOutput data, I have a third PVOutput account for the combined data, which is filled using the python solution [CombinePVOutputSystems](https://github.com/ZuinigeRijder/CombinePVOutputSystems#combine-pvoutput-systems).

My first response would be to have 2 scripts running in different directories (one for each inverter) and for the second directory you do modifications, e.g. the python script to get the second inverter.
Create two directories, e.g. directory solis and directory solis2 and fill them both with the solution.

In solis2 directory you change the following:
- modify soliscloud_to_pvoutput.cfg to point the second PVOutput account secrets
- rename solis.sh to solis2.sh and modify solis2.sh to go to directory solis2 (line 9: cd ~/solis2)
- modify soliscloud_to_pvoutput.py to get the second inverter (line 168, change the 0 into 1)
- have two cronrabs running (for solis.sh and solis2.sh)

Modify From:
````
    station_info = json.loads(content)["data"]["page"]["records"][0]
````
into:
````
    station_info = json.loads(content)["data"]["page"]["records"][1]
````

Let me know if this approach works for you. Maybe I can have the inverter index configurable in the future, so at least you do not have to modify the soliscloud_to_pvoutput.py script. But of course, I need to be sure the above is working, so I can also add this information to the README.

**Reply by Skjal** [reply]
Hi. Thanks for the suggestion. I think this would work but haven't had the time today to set it up. Would try to do this tomorrow and report back. This would be better than the solution I currently using since I then can track both individually in domoticz.

**Comment by Skjal** [comment]
Hi. Got it working following your guide. The only thing that i needed to change was this "modify soliscloud_to_pvoutput.py to get the second inverter (line 168, change the 0 into 1)" My inverters was under the same system so got an error while trying to change this line but I changed line 173 instead to this "inverter_info = json.loads(content)["data"]["page"]["records"][1]" and then it worked. :) hope this makes sense to you, if not I will try to explain better next time. 
 Also, to be able to have two outputs in pvoutput you need to donate to pvoutput or else you would not be able to make another system.

**Comment by ZuinigeRijder** [comment]
@Skjal You are right, I did mention and copied the wrong line, it must be indeed line 173:

````
inverter_info = json.loads(content)["data"]["page"]["records"][1]
````

> Also, to be able to have two outputs in pvoutput you need to donate to pvoutput or else you would not be able to make another system.

You do not need to donate to PVOutput, just make 2 PVOutput accounts (you need 2 email addresses). The second account you also configure the API key and for solis2 you configure the second PVOutput account as target. 

And as posted earlier, if you also want the combined data of the two inverters, use a third PVOutput account (yet another email address) and use my python tool [CombinePVOutputSystems](https://github.com/ZuinigeRijder/CombinePVOutputSystems#combine-pvoutput-systems).

I will make the inverter index configurable and will update the README.md to incorporate this information mentioned in this discussion.

**Comment by ZuinigeRijder** [comment]
I published a [new release R1.11.0](https://github.com/ZuinigeRijder/SolisCloud2PVOutput/releases/tag/R1.11.0)

**Comment by Skjal** [comment]
Thanks, will check it out. Seems like I skipped the part where you explained about pvoutput. That's the reason that I thought you needed to donate to make this work. Anyway lesson learned. :).  Next year when my donation runs out I will implement your solution for pvoutput but until then I will make use of my donation. :)
